app/models/nic/base.rb

```diff
@@ -149,6 +150,14 @@ def normalize_mac
       self.errors.add(:mac, e.message)
     end
 
+    def valid_domain
+      begin
+        Domain.find(domain_id)
+      rescue
```

Comment:
we shouldn't use exceptions here, let's do `Domain.find_by_id(domain_id)` and add errors only if it's nil